pkg/install/stack/kubermatic-master/stack.go

```diff
@@ -380,6 +380,12 @@ func showDNSSettings(ctx context.Context, logger *logrus.Entry, kubeClient ctrlr
 		logger.Infof("   %s.    IN  CNAME  %s.", domain, hostname)
 		logger.Infof("   *.%s.  IN  CNAME  %s.", domain, hostname)
 	} else if ip := ingresses[0].IP; ip != "" {
+		for _, ingress := range ingresses {
+			if common.ValidatePublicIp(ingress.IP) {
+				ip = ingress.IP
+				break
+			}
+		}
```

Comment:
Thinking more about this, I think this needs to be rewritten a bit more. If we're now looking through all ingresses for a suitable (or at least better) IP, but further up we also clearly prefer a _hostname_, then why should we not also look through all ingresses for a hostname, instead of just the first one?

This should be rewritten to something similar to

```
var hostname, ip string
for all ingresses {
   if has hostname { hostname = ingress.hostname }
   if has ip {
      // prefer a public IP, but private IPs are also okay depending on the KKP setup
      if ip == "" { ip = ingress.ip }
      else if is public ingress.ip { ip = ingress.ip }
   }
}

if hostname { log log log }
else if ip { log log log }
```